src/backends/mysql/session.cpp

```diff
@@ -335,6 +337,15 @@ void parse_connect_string(const string & connectString,
             char *endp;
             *write_timeout = std::strtoul(val.c_str(), &endp, 10);
             *write_timeout_p = true;
+        } else if (par == "ssl_mode" && !*ssl_mode_p)
+        {
+            if (val=="DISABLED") *ssl_mode = SSL_MODE_DISABLED;
+            else if (val=="PREFERRED") *ssl_mode = SSL_MODE_PREFERRED;
+            else if (val=="REQUIRED") *ssl_mode = SSL_MODE_REQUIRED;
+            else if (val=="VERIFY_CA") *ssl_mode = SSL_MODE_VERIFY_CA;
+            else if (val=="VERIFY_IDENTITY") *ssl_mode = SSL_MODE_VERIFY_IDENTITY;
+            else throw soci_error(err);
```

Comment:
I think it would be nice to say that "ssl mode is invalid" here instead of giving just the generic error message.

Reply:
Hi,

As you suggested I've fixed the `soci_error()` call to provide a relevant string, and I have modified the docs to describe the addition.

The `mysql_ssl_set()` call I don't understand - it occurs outside the bit I have touched, at line 420 (my version). I used `mysql_options()` (line 459) to do my work. Can't see how I goofed...